lib/puppet/vendor/semantic_puppet/lib/semantic_puppet.rb

```diff
@@ -1,15 +1,7 @@
-begin
-  require 'gettext-setup'
-rescue LoadError
-  def _(msg)
-    msg
-  end
-end
+require 'gettext-setup'
 
 module SemanticPuppet
-  if defined?(GettextSetup)
-    GettextSetup.initialize(File.absolute_path('semantic_puppet/locales', File.dirname(__FILE__)))
-  end
```

Comment:
Strictly speaking, we don't need it in the gem, because the gem can/does express a dependency on gettext, so it should always be satisfied when running as a gem. The reason we need the gettext check specifically here in the vendored case is because this puppet code can be loaded from a Puppet Server install pre-2.7.2 that does not have gettext available - so both here, and in puppet (https://github.com/puppetlabs/puppet/blob/master/lib/puppet.rb#L28-L37) we do this check.  It was definitely not ideal to add it into the vendored lib as it represents a divergence from the upstream library but it seemed the least of all evils. We want to find a better solution, and have that ticketed. These issues: https://tickets.puppetlabs.com/browse/PA-1018, https://tickets.puppetlabs.com/browse/PUP-7114, and https://tickets.puppetlabs.com/browse/PUP-7116 have more context.

My recommendation here, as unfortunate as it is, is to just patch this vendored copy to do the previous behavior prior to this PR - ie check for gettext and only load if necessary. I don't think the same changes are needed in the gem.